### DISCUSSION TITLE
Forum as Documentation  <---o---> Forum to Documentation process

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#261] [General]
Hi all,
I was thinking about **Freeplane's documentation** and came to this **thoughts and ideas**:

- The gap between the documentation and all the features of the actual version of Freeplane is considerable
- Closing this gap will be a big amount of work that can not be made by a single person or just a few persons in their spare time
- There are potentially a lot of people willing to help but with little tasks that are clear defined and abordable and not asuming a long term responsability
- A big part of this people are not confident in english, but can help if somebody can translate or correct their writings
- Until now, most of the individual efforts have not reached the desired impact. This can be very frustrating because it has been a lot of time and thinking and reviewing and learning and discussing and ...
- Freeplane is a multipurpose tool with a lot of flexibilty that allows lots of different uses and workflows, This make very difficult to think in a tutorial or documentation that  can cover all features and use cases
- Freeplane is been actively developed and it changes and adds new features constantly. That's good for the users, but the documentation gets more outdated day after day

### How has the community bypassed this situation till now? 

---> By using the **forum as a central documentation** source

- Much of the missing information in the documentation has been written as posts in the forum (great!!)
- To find that information its pretty difficult  (at least in the SourceForge forum)
- To filter between different Topics is also very difficult and inaccurate
- We haven't been able to transfer that new and precious information to the documentation in order to keep it up to date
- The actual SourceForge forum doesn't give us tools to help us with that task

Those are mostly the reasons that I have to take a chance with the GitHub forum.
But to get the benefits of making the effort of changing the forum platform, then we have to 

- get confident with the tools that offers the GitHub Discussions platform
- find a way to easily categorize the posts so they can be easily searched and filtered when looking for a specific information about a feature or a use case, etc.
- find a way in which many people can help in this task but that still has an order and follows a standard

I think using categories and labels can be helpful in this task.

- But how to define the available categories and labels is something that we have to discuss. Surely we can look at other FOSS forums and learn from their experiences.
- How and who applies the labels to the posts is another problem. I read a little in the documentation and it is not open to everyone. The user needs some privileges to assign labels to a discussion, but that privileges allow other tasks that reduce the number of persons that will be able to help. 
- There is the option to use GitHub Apps or Bots that following some defined rules can apply labels and make some predefined tasks automatically. I think this is very interesting and we have to inform us about their capabilities.

Please, if somebody knows more about those topics, then share your experience in a post. Thanks!

If we can get this labeling thing to work, then, I think, we should be able to master our "**Forum as Documentation**" situation and hopefully implement a successful "**Forum to Documentation**" process next.

Have a great day,

edo

**Comment by EdoFro** [comment]
Some of the sources about managing discussions:

- [Managing categories for discussions in your repository](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-categories-for-discussions-in-your-repository)
- [Managing labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels)
  - [Applying a label](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#applying-a-label)
  "Anyone with **triage** access to a repository can apply and dismiss labels."
- [Repository roles for an organization](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization)
  - [Permissions for each role](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role)

other basic info

- [GitHub Discussions](https://docs.github.com/en/discussions)
- [Quickstart for GitHub Discussions](https://docs.github.com/en/discussions/quickstart)


GitHub Apps and Bots

- lots of information there that I haven't been able to read yet

**Reply by EdoFro** [reply]
Just a little preview about  GitHub Robots:

YouTube 5 mins video: [WRITING BOTS FOR GITHUB](https://www.youtube.com/watch?v=cMenLkK8Ulc)

**Comment by rexelbartolome** [comment]
This is interesting, never thought about this way. Might not be the best solution but making the forums more search-friendly is definitely a net positive.

I'd be wary of categories though. Tags/labels are okay. I think splitting the discussion with categories (like in SF where you have to click inside Documentation Writers discussion to see whats up there) would thin out the already relatively small community (and discussions) we have. I think categories should only be considered once the freeplane community has grown a fair bit. Tho im not even sure if that kind of category function is the one used in github discussions, afaik you can only filter different tags and labels.

I also think there should be an in-program thing to point new users to github discussions if they're looking for help. And maybe mention it in the documentation/wiki that asking a question over here is also an option if they cant find an answer in the wiki.

**Reply by quickfold** [reply]
edit: sorry, this was posted as a reply to @rexelbartolome but was intended to reply to OP.

Thanks for the post @EdoFro. Here are my comments.

> Closing this gap will be a big amount of work that can not be made by a single person or just a few persons in their spare time

Significantly reducing this gap is still potentially valuable even if it can't be totally eliminated, and that could  be done by a few people in their spare time, but maybe only over a long time period.

> A big part of this people are not confident in english, but can help if somebody can translate or correct their writings

I'll say it again—it doesn't matter if the English is not perfect as long as it gets the point across. I hope non-native speakers can accept this and not worry too much about the quality of their grammar.

> Freeplane is been actively developed and it changes and adds new features constantly. That's good for the users, but the documentation gets more outdated day after day

And unfortunately, this has been the case for 10 years, so the documentation is way out of date.

Now, as far as making the forum more useful with better tags and labels, this is potentially useful. One possible problem is that if no one has responsibility for tagging or labeling posts, and/or if bots can't do it, then people might not trust that the tags will actually find all of the posts they need. And if the tags are not good enough to trust, people might use them but not find what they need, or they might not use them and just default to text search to find what they want, in which case tagging then doesn't add value. I'm not sure the best way to approach this issue.

**Comment by dpolivaev** [comment]
Different people need different kinds of documentation and support. The needs of nerds and power users are different from the needs of absolute beginners. The amount and quality of documentation should fit the reader.

I see three different approaches to achieving it which can also be combined with each other.

1. There are could be different sources of documentation so that people can find what fits their needs better. And actually, it is nearly the current situation with different videos, articles, mind maps, wiki pages, forum posts, and so on. Here I think a good up-to-date overview of the existing sources could probably help as well as new documentation, and any kind of documentation had its chance to find its reader.
2. New documentation could be written with a concrete kind of user in mind, e.g. beginner, script developer, project manager, and so on.
3. If you could share parts of your maps they could also be used as tutorials.

Because all paths could be beneficial you can decide which one you want to try.

**Reply by quickfold** [reply]
I agree that the current situation is that there are different sources of documentation. This does not seem to be working very well. Even if someone puts together an overview, what exists is not sufficient to document Freeplane.

If there is not a common documentation goal and path towards that goal specified through conversation and or leadership, then documentation will advance only by individuals trying what seems best to them individually. It does not seem that people have been interested in going that route, for the most part. Encouraging individual contributions has not produced significant improvements in documentation for the last 10 years.

Lots of people contributing little by little can produce great things but only if there is a clearly conceived structure to guide their contributions, as in the case of Wikipedia. If Wikipedia just had people try different things as they liked but there was no top-down strategy, it is hard for me to imagine it being successful.
